File: rider-fsharp/src/main/java/com/jetbrains/resharper/ideaInterop/fileTypes/fsharp/FSharpLanguage.kt
package com.jetbrains.resharper.ideaInterop.fileTypes.fsharp

import com.jetbrains.resharper.ideaInterop.fileTypes.RiderLanguageBase

object FSharpLanguage : RiderLanguageBase("F#", "FSHARP") {
    override fun isCaseSensitive() = true
}

File: ReSharper.FSharp/src/FSharp/FSharp.ProjectModelBase/src/FSharpLanguage.cs
using JetBrains.Annotations;
using JetBrains.ReSharper.Psi;

// ReSharper disable once CheckNamespace
namespace JetBrains.ReSharper.Plugins.FSharp.Psi
{
  [LanguageDefinition(Name)]
  public class FSharpLanguage : KnownLanguage
  {
    public new const string Name = "F#";

    [CanBeNull, UsedImplicitly]
    public static FSharpLanguage Instance { get; private set; }

    public FSharpLanguage() : base(Name)
    {
    }

    protected FSharpLanguage([NotNull] string name) : base(name)
    {
    }

    protected FSharpLanguage([NotNull] string name, [NotNull] string presentableName) : base(name, presentableName)
    {
    }
  }
}
